### PR TITLE
Add tests for guest agent data

### PR DESCRIPTION
**Pull request by pcbailey**
This PR adds test for the guest agent data added to the Overview and Details tabs.

### CODE REVIEW

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+  NO_ACTIVE_USERS,
+} from './utils/constants/common';
+import { VirtualMachine } from './models/virtualMachine';
+import { getWindowsVM } from './utils/templates/windowsVMForRDPL2';
+import { multusNAD } from './mocks/mocks';
+
+const VM_NAME = 'windows-rdp';
+const VM_IP = '123.123.123.123';
+
+describe('KubeVirt VM console - RDP', () => {
```
Comment (rhrazdil) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts:44]: This should be more general. Like `Tests involving guest agent`, but maybe you can think of something better

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+  let vm: VirtualMachine;
+
+  beforeAll(async () => {
+    createResource(multusNAD);
+
+    // for cmd-line scripts only
+    execSync(`kubectl config set-context --current --namespace=${testName}`);
+    execSync('kubectl create -f -', {
+      input: getWindowsVM({
```
Comment (rhrazdil) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts:53]: Could you please rename the method to `getFakeWindowsVM`?

Comment (pcbailey) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts]: Done!

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+    execSync(`kubectl config set-context --current --namespace=default`);
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = originalDefaultTimeout;
+  });
+
+  describe('Testing guest agent data', () => {
+    it('Displays guest agent data in Details tab', async () => {
+      await vm.navigateToDetail();
+
+      await browser.wait(
+        waitForStringInElement(vmView.vmDetailTimeZone(testName, VM_NAME), 'UTC'),
```
Comment (rhrazdil) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts:74]: If we do this in beforeAll, then we don't need it here

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+        VM_CREATE_AND_EDIT_AND_CLOUDINIT_TIMEOUT_SECS,
+      );
+
+      expect(vmView.vmDetailHostname(testName, VM_NAME).getText()).toContain(VM_NAME);
+      expect(vmView.vmDetailTimeZone(testName, VM_NAME).getText()).toEqual('UTC');
+
+      await browser.wait(
+        until.presenceOf(vmView.vmDetailActiveUsersListNoUsers),
```
Comment (rhrazdil) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts:82]: question: Should the active users list be displayed immediately when the page loads or is it acceptable id it's displayed after some time?
In any case, 15 minutes sounds like a lot

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+      await vm.navigateToOverview();
+      expect(dashboardView.vmDetailsHostname.getText()).toContain(VM_NAME);
+      expect(dashboardView.vmDetailsTZ.getText()).toContain('UTC');
+      expect(dashboardView.vmDetailsLoggedInUsers.getText()).toEqual('No users logged in');
+    });
+
+    it('Displays guest agent data in Disks tab', async () => {
+      await vm.navigateToDisks();
+      await browser.wait(
+        until.presenceOf(disksView.fileSystemsTableHeader),
```
Comment (rhrazdil) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts:99]: similar as above

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+
+  afterAll(async () => {
+    deleteResource(multusNAD);
+    deleteResource(vm.asResource());
+    execSync(`kubectl config set-context --current --namespace=default`);
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = originalDefaultTimeout;
+  });
+
+  describe('Testing guest agent data', () => {
+    it('Displays guest agent data in Details tab', async () => {
```
[the remark this answers is not in the document]
Comment (gouyang) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts:70]: I will create these TCs.

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+const VM_NAME = 'windows-rdp';
+const VM_IP = '123.123.123.123';
+
+describe('Tests involving guest agent', () => {
+  let vm: VirtualMachine;
+  const originalDefaultTimeout = jasmine.DEFAULT_TIMEOUT_INTERVAL;
+
+  beforeAll(async () => {
+    // Extending default timeout due to time required for guest agent to load
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = VM_WITH_GA_CREATE_AND_EDIT_CLOUDINIT_TIMEOUT_SECS;
```
[the remark this answers is not in the document]
Comment (pcbailey) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts]: Wait moved to `beforeAll`. Thanks for the tip!

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts (hunk 3 above)
Comment (pcbailey) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts]: Done!

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts (hunk 4 above)
Comment (pcbailey) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts:80]: There was a delay when it changed pages. You're definitely right about the timeout length, though. I created a new timeout for it and set it to 30 seconds.

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts (hunk 5 above)
Comment (pcbailey) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts:97]: Done!

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+import { multusNAD } from './mocks/mocks';
+
+const VM_NAME = 'windows-rdp';
+const VM_IP = '123.123.123.123';
+
+describe('Tests involving guest agent', () => {
+  let vm: VirtualMachine;
+
+  beforeAll(async () => {
+    // Extending default timeout due to time required for guest agent to load
```
Comment (rhrazdil) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts:46]: ops, could you remove the comment please? :)

**Review by rhrazdil — CHANGES_REQUESTED** [review]

File: frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts (hunk 8 above)
Comment (pcbailey) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/vm.guest.agent.scenario.ts]: Good catch! =) Fixed!

**Comment by rhrazdil** [comment]
Thanks!
/lgtm
/approve

**Pull request merged**